gnocchi/storage/ceph.py

```diff
@@ -202,7 +202,9 @@ def _pending_measures_to_process_count(self, metric_id):
         return len(list(self._list_object_names_to_process(object_prefix)))
 
     def list_metric_with_measures_to_process(self, size, part, full=False):
-        names = self._list_object_names_to_process()
+        names = sorted(
+            self._list_object_names_to_process(),
+            key=lambda x: '{}-{}'.format(x.split('_')[3], x.split('_')[4]))
```

Comment:
I'm not sure it's worth replacing the `_` by `-` between `[3]` and `[4]` so you can just do:

`lambda x: x.split("_", 3)[-1]`